syntax = "proto3";
package A3;
message Book {
    string ISBN = 1;
    string title = 2;
    string author = 3;
    enum Genre {
        FICTION = 0;
        MYSTERY = 1;
        HISTORY = 2;
        POETRY = 3;
    }
    int32 publishing_year = 4;
}

message InventoryItem {
    int32 inventory_num = 1;
    oneof item_type {
        Book book = 2;
    }
    enum Status {
        AVAILABLE = 0;
        TAKEN = 1;
    }
}